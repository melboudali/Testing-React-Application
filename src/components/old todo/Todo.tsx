interface todoType {
	todo: { id: number; title: string; completed: boolean };
}
const Todo = ({ todo: { id, title, completed } }: todoType) => {
	const h1 = <h1>{title}</h1>;
	const text = completed ? <del>{h1}</del> : h1;
	return <div data-testid={`todo-${id}`}>{text}</div>;
};

export default Todo;
