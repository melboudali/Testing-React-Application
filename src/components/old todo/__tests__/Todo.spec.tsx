import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "../Todo";

afterEach(() => {
	cleanup();
});

// it("should render todo component", () => {
// 	render(<Todo />);
// 	const todoElement = screen.getByTestId("todo-1");
// 	expect(todoElement).toBeInTheDocument();
// 	expect(todoElement).toHaveTextContent("Hello World!");
// });

it("should render non-completed todo", () => {
	const todo = { id: 1, title: "wash dishes", completed: false };
	render(<Todo todo={todo} />);
	const todoElement = screen.getByTestId("todo-1");
	expect(todoElement).toBeInTheDocument();
	expect(todoElement).toHaveTextContent("wash dishes");
});

it("should render completed", () => {
	const todo = { id: 2, title: "make dinner", completed: true };
	render(<Todo todo={todo} />);
	const todoElement = screen.getByTestId("todo-2");
	expect(todoElement).toBeInTheDocument();
	expect(todoElement).toHaveTextContent("make dinner");
	expect(todoElement.firstChild).toHaveStyle("text-decoration: line-through;");
});

it("matches snapshot", () => {
	const todo = { id: 1, title: "wash dishes", completed: false };
	const tree = renderer.create(<Todo todo={todo} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("matches snapshot without react-test-renderer", () => {
	const todo = { id: 1, title: "wash dishes", completed: false };
	const { container } = render(<Todo todo={todo} />);
	expect(container).toMatchSnapshot();
});

export {};
