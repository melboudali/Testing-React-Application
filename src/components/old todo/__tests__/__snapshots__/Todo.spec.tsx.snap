// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;

exports[`matches snapshot without react-test-renderer 1`] = `
<div>
  <div
    data-testid="todo-1"
  >
    <h1>
      wash dishes
    </h1>
  </div>
</div>
`;
